UI/src/i18n.js

```diff
@@ -15,42 +16,53 @@ function _mapLocale(locale) {
     return locale;
 }
 
+var _messages = {};
+SUPPORT_LOCALES.forEach(function (it) {
+    _messages[it] = require("./locales/" + it + ".json");
+});
+
 const i18n = createI18n({
     globalInjection: true,
+    useScope: "global",
     legacy: false,
     fallbackWarn: false,
     missingWarn: false, // warning off
     locale: _mapLocale(window.lsmbConfig.language),
     fallbackLocale: "en",
-    messages: {
-        en: require("./locales/en.json")
-    }
+    messages: _messages
 });
 
+export function setI18nLanguage(locale) {
+    if (i18n.mode === "legacy") {
+        i18n.global.locale = locale;
+    } else {
+        i18n.global.locale.value = locale;
+    }
+    document.querySelector("html").setAttribute("lang", locale);
+    if (rtlDetect.isRtlLang(locale)) {
+        document.querySelector("html").setAttribute("dir", "rtl");
+    }
+}
+
 export async function loadLocaleMessages(locale) {
-    const _locale = _mapLocale(locale);
+    let _locale = _mapLocale(locale);
     if (SUPPORT_LOCALES.includes(_locale)) {
         // load locale messages
-        if (!i18n.global.availableLocales.includes(locale)) {
+        if (!i18n.global.availableLocales.includes(_locale)) {
             // load locale messages with dynamic import
             const messages = await import(
                 /* webpackChunkName: "locale-[request]" */ `./locales/${_locale}.json`
             );
-
             // set locale and locale messages
-            i18n.global.setLocaleMessage(locale, messages);
-        }
-        // Update document
-        document.querySelector("html").setAttribute("lang", locale);
-
-        // Switch the whole application to this locale
-        if (i18n.mode === "legacy") {
-            i18n.global.locale = locale;
-        } else {
-            i18n.global.locale.value = locale;
+            i18n.global.setLocaleMessage(_locale, messages);
         }
+    } else {
+        _locale = "en";
     }
+    setI18nLanguage(_locale);
     return nextTick();
 }
 
+loadLocaleMessages(window.lsmbConfig.language);
```

Comment:
Since this function is `async`, shouldn't this be prefixed with "await"?

Also: this duplicates the `beforeCreate` handler in `main-vue.js`. I'm thinking that, since the `beforeCreate` handler can't be `async`, we probably want to remove that invocation in favor of this one?

Reply:
Duplicated code removed. `beforeCreate` could have been made async though.

Reply:
I see that's how it's been resolved now. Thanks! No further remarks or nits. Only praise! :+1: